src/lib/components/Widget.tsx

```diff
@@ -27,9 +29,14 @@ export interface WidgetProps {
   children: ReactNode
   theme?: Partial<Theme>
   locale?: SupportedLocale
+  provider?: EthProvider
 }
 
-export default function Widget({ children, theme, locale = DEFAULT_LOCALE }: WidgetProps) {
+export default function Widget({ children, theme, locale = DEFAULT_LOCALE, provider }: WidgetProps) {
+  const [, setProvider] = useAtom(providerAtom)
+  useEffect(() => {
+    setProvider(provider)
+  }, [setProvider, provider])
```

Comment:
Yes, but we typically include even referentially stable hook dependencies:
`React Hook useEffect has a missing dependency: 'setProvider'. Either include it or remove the dependency array.eslintreact-hooks/exhaustive-deps`

For example, we usually include `dispatch` when it is used, even though that's not needed. I'd rather include unneeded dependencies than override w/ `disable-next-line` when I can, since doing so might lead to unintentional omissions.

Reply:
I'm definitely in favor of just included unneeded deps over adding eslint overrides.